rest/src/main/java/org/jboss/pnc/rest/SecurityConstraintFilter.java

```diff
@@ -52,32 +52,17 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
         String path = requestContext.getUriInfo().getPath();
 
         if ((path.matches("/builds/ssh-credentials.*") || path.matches("/users/current.*"))
-                && Strings.anyStringEquals(method, HttpMethod.GET)
-                && !isUserInAnyRole(UserService.ROLE_USER, UserService.ROLE_ADMIN, UserService.ROLE_SYSTEM_USER)) {
+                && Strings.anyStringEquals(method, HttpMethod.GET) && !userService.isUserLoggedIn()) {
             throw new NotAuthorizedException("Authorization required to access this resource.");
         }
-        if (path.matches("/.*")
-                && Strings.anyStringEquals(
-                        method,
-                        HttpMethod.POST,
-                        HttpMethod.PUT,
-                        HttpMethod.DELETE,
-                        HttpMethod.HEAD,
-                        HttpMethod.PATCH)
-                && !isUserInAnyRole(UserService.ROLE_USER, UserService.ROLE_ADMIN, UserService.ROLE_SYSTEM_USER)) {
+        if (path.matches("/.*") && Strings.anyStringEquals(
+                method,
+                HttpMethod.POST,
+                HttpMethod.PUT,
+                HttpMethod.DELETE,
+                HttpMethod.HEAD,
```

Comment:
Just for curiosity, why do we require user to be authenticated for HEAD request? I see that it has been here since forever but it strikes like a mistake.

Reply:
No I don't think it should be there, good catch! I can create another PR to fix that.

Reply:
Nice!

Reply:
https://github.com/project-ncl/pnc/pull/3877